kontakt-reservasjonsregister-client/src/main/java/no/difi/meldingsutveksling/ptp/OppslagstjenesteClient.java

```diff
@@ -1,71 +1,130 @@
         this.conf = configuration;
     }
 
-    public KontaktInfo hentKontaktInformasjon(String pid) {
+    /**
+     * Brukes til å hente parametre til sikker digital post forsendelse
+     *
+     * @param lookupParameters parametre til KRR oppslaget, dvs. personnummeret og virksomhetens organisasjonsnummer
+     * @return parametre for å sende sikker digital post
+     */
+    public KontaktInfo hentKontaktInformasjon(LookupParameters lookupParameters) {
         final HentPersonerForespoersel hentPersonerForespoersel = HentPersonerForespoersel.builder()
                 .addInformasjonsbehov(Informasjonsbehov.KONTAKTINFO, Informasjonsbehov.SIKKER_DIGITAL_POST, Informasjonsbehov.SERTIFIKAT, Informasjonsbehov.VARSLINGS_STATUS)
-                .addPersonidentifikator(pid)
+                .addPersonidentifikator(lookupParameters.getIdentifier())
                 .build();
 
         WebServiceTemplate template = createWebServiceTemplate(HentPersonerRespons.class.getPackage().getName());
 
-        final HentPersonerRespons hentPersonerRespons = (HentPersonerRespons) template.marshalSendAndReceive(conf.url, hentPersonerForespoersel);
+        WebServiceMessageCallback callback;
+        if (conf.isPaaVegneAvEnabled()) {
+            callback = new SoapHeaderCallbackHandler(lookupParameters.getClientOrgnr()); // must be added in production
+        } else {
+            callback = noopCallback -> {}; // only to be used in test/development.
+        }
+        final HentPersonerRespons hentPersonerRespons = (HentPersonerRespons) template.marshalSendAndReceive(conf.url, hentPersonerForespoersel, callback);
 
         return KontaktInfo.from(hentPersonerRespons);
 
     }
 
     /**
-     *  Tjenesten kan brukes for å sende forsendelser av brev til mottakere som har reservert seg eller ikke registrert/oppdatert sin kontaktinformasjon.
+     * Tjenesten kan brukes for å sende forsendelser av brev til mottakere som har reservert seg eller ikke registrert/oppdatert sin kontaktinformasjon.
+     *
+     * @param lookupParameters inneholder organisasjonsnummeret til virksomheten man gjør oppslag på vegne av
+     * @return parametre for å sende melding til printtjenesten
      */
-    public PrintProviderDetails getPrintProviderDetails() {
+    public PrintProviderDetails getPrintProviderDetails(LookupParameters lookupParameters) {
         HentPrintSertifikatForespoersel request = new HentPrintSertifikatForespoersel();
         WebServiceTemplate template = createWebServiceTemplate(HentPrintSertifikatRespons.class.getPackage().getName());
-
-        HentPrintSertifikatRespons response = (HentPrintSertifikatRespons) template.marshalSendAndReceive(conf.url, request);
+        WebServiceMessageCallback callback;
+        if (conf.isPaaVegneAvEnabled()) {
+            callback = new SoapHeaderCallbackHandler(lookupParameters.getClientOrgnr());
+        } else {
+            callback = noopCallback -> {}; // only needed during development/test.
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either remove or fill this block of code. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS00108)